### ISSUE TITLE
Creation of questionnaire

### ISSUE THREAD

**Issue by froggypaule** [ds-wizard/ds-wizard#202]
**The Question**
adding questions to a questionnaire

**DS Wizard part**
Questionnaire

**Additional context**
Hello Marek. There is something I don't quite grasp: when one creates a questionnaire from a km, one is asked to add the questions: 
1. I don't see why this choice is offered: I would have added by default all the questions as per how the km is structured
2. Also, at this stage, we have lost all general overview of the km making it difficult to remember which question is actually was

Likewise, when a researcher answers a questionnaire, they also may add questions.
In short, I must have missed something: what is the rationale behind this feature? 
Thanks!

**Comment by MarekSuchanek** [comment]
Hi @froggypaule, I probably don't really understand the question... You mean "Add" for list question? It does not add questions but "items" where are follow-up questions for each item based on definitions from the KM.

**Comment by froggypaule** [comment]
Good morning Marek. I hope you had a good rest (remember the DSW announcement at the end of the year).
I join two pics to explain myself : the first is that of the expanded km, the second is that of the newly created questionnaire. 
![expanded_km](https://user-images.githubusercontent.com/47455219/104157263-1b937280-53eb-11eb-8020-efe5ef352f15.png)
![questionnaire](https://user-images.githubusercontent.com/47455219/104157271-20582680-53eb-11eb-8e11-594ae21b5bb5.png)

What I mean is why does the questionnaire not include the questions by default, to perfectly align with the km?
Also, when adding the question (or not), one has no nice overview of the structure of the km.

Does this make sense? sorry if I can't explain it better....

**Comment by MarekSuchanek** [comment]
It is like this because those three questions are "list questions", instead of that you can make "options question" with single option "fill" (to keep the structure). But there is no "tree-view" pane in questionnaire.

**Comment by froggypaule** [comment]
Ok, I see. But what is the reason for not listing all items directly in the questionnaire? I must miss something. Thanks.

**Comment by MarekSuchanek** [comment]
List question is intended for user to enumerate something and give details about each item of that enumeration. For example, question "What software do you use?" could have following "item" questions "What is the name of that software?", "What is the license?", etc. So then I can click "Add" per each software I use and fill... e.g. "DSW" for name and "Apache 2" for license in the first item, and then "Libre Office" and "Mozilla Open License" for the second, etc.

It is also possible that someone is not using any software and in that care the user won't press "Add" and there will be no item.

**Comment by froggypaule** [comment]
Ok, I see: I understand where you are coming from. Still, could we say that it would be good idea --in general-- to be able to expand the km when filling the questionnaire?. I will look into the 'ideas' page. Many thanks.

**Comment by froggypaule** [comment]
Hello Marek: actually I reopen this issue. Going back to this 'add' thing. I and the users find it confusing:
1. before 'adding', one does not see the questions we are about to add, so we are a bit blind
2. is there a way to restrict the number of times you can add a list of items: what I mean by that, most of the time you need only add one set of items, and that's it

And no, I don't think options are the solutions here. 
One use case:
- the labelling of the list of items question is: 'Description of the dataset'
- there follows a bunch of item questions
- but I only need to add this list once
- and a list of options is not suitable

possibly this question has a relationship with an idea I saw mentionned asking for a expansion of the structure beyond chapter.

anyhow, I hope I make sense here. Thanks.

**Comment by MarekSuchanek** [comment]
Hello Paulette,

1. Yes, it is intentional as the "subquestions" should be context-related to the specific "item". For example, "List datasets you use"... you are not interested what we ask about each dataset if you know that you don't use any of them. 
2. That is not possible... maybe in some use cases it makes sense (please feel free to create a feature request in ideas.ds-wizard.org). But for 0 and 1, I still think that having option "Do you use a dataset?" with Yes/No options is better instead of list question.

**Comment by froggypaule** [comment]
1. ok I see: this then implies to rephrase of the questions to clarify this point (rephrase them with respect to my skewed and current understanding). This btw, partially answers 2. as well. I say 'partially' because I am not (yet) quite convinced by this view of the DMP. In case something precise comes in mind, I'll log it in 'ideas'.
Thanks!
